### ISSUE TITLE
Rule S1854: Rule: Dead stores should be removed

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-dotnet#2050]
### Description

Implements https://jira.sonarsource.com/browse/RSPEC-1854

**Comment by Evangelink** [comment]
This requires a CFG so let's remove the rule from the current sprint.

**Comment by andrei-epure-sonarsource** [comment]
I don't think we plan to add a CFG to VB.net analyzer too soon, right @nicolas-harraudeau-sonarsource  - I may close this ticket.

**Comment by sm-g** [comment]
I have false positive:

```
var templateName = string.Empty;
try
{
	var sshTemplate = ParseSshTemplate(template);
	templateName = sshTemplate.TemplateName; // this assignment need
	return null;
}
catch (Exception e)
{
	_eventsService.RaiseEvent(new Event(e, templateName));
	throw;
}
```

**Comment by andrei-epure-sonarsource** [comment]
we won't add a CFG to VB.NET anytime soon.

**Comment by andrei-epure-sonarsource** [comment]
@sm-g - please open a new issue, specifying the version of SonarCSharp you are using and the version of MSBuild - this ticket was tracking implementing this rule for the VB.NET analyzer
